### PR TITLE
CLN: Refactor tests and delete duplicates

### PR DESCRIPTION
**Pull request by phofl**
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

Deleted a few duplicates and parametrized

### CODE REVIEW

File: pandas/conftest.py [hunk 1]
```diff
@@ -1598,6 +1598,14 @@ def indexer_sl(request):
     return request.param
 
 
+@pytest.fixture(params=[tm.at, tm.loc])
+def indexer_sa(request):
```
Comment (jbrockmendel) [inline pandas/conftest.py:1602]: maybe `indexer_al` or `indexer_la`?

Comment (phofl) [reply pandas/conftest.py]: Ok with me, added the s because of the setitem

Comment (jreback) [reply pandas/conftest.py]: yeah I would ideally make these a bit more descriptive (all of them), at the expense of slightyl increasing length :-> (but let's do this as a followon).

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 2]
```diff
@@ -412,13 +409,6 @@ def test_loc_getitem_duplicates_multiindex_missing_indexers(indexer, pos):
         tm.assert_series_equal(result, expected)
 
 
-def test_series_loc_getitem_fancy(multiindex_year_month_day_dataframe_random_data):
```
Comment (jbrockmendel) [inline pandas/tests/indexing/multiindex/test_loc.py:415]: this is duplicated somewhere?

Comment (phofl) [reply pandas/tests/indexing/multiindex/test_loc.py:415]: Yes, was tricky to find.

test_loc_getitem_list_of_tuples_with_multiindex same file line 81. Different names but content is the same

File: pandas/conftest.py (hunk 1 above)
Comment (phofl) [reply pandas/conftest.py]: Will put up something after this is merged

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/indexing/multiindex/test_loc.py:415]: nice.  these can be tough, but always fulfilling to rip out

**Comment by jreback** [comment]
thanks @phofl

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 2 above)
Comment (phofl) [reply pandas/tests/indexing/multiindex/test_loc.py:415]: Yes definitely :)

**Pull request merged**